include/deal.II/dofs/dof_objects.h

```diff
@@ -204,7 +204,7 @@ namespace internal
     {
       (void)fe_index;
       Assert(
-        (fe_index == dealii::DoFHandler<dh_dim, spacedim>::default_fe_index),
+        (fe_index == 0),
```

Comment:
The static variable `default_fe_index` used to have the value `0` in the case of `DoFHandler` and `-1` in the case of `hp::DoFHandler`. This is not possible anymore. What is a solution call give different names, e.g. `default_fe_index` and `invalid_fe_index`? @marcfehling What is your take on this?

Reply:
I don't see the point of keeping the `default_fe_index` variable any longer if we merge both `DoFHandlers`. The regular `DoFHandler` only allows one fe, and all cells will be initialized with the zero `active_fe_index` in hp mode. So why not hard code it to the zeroth index as you did, and distinguish between regular and hp mode with the `is_hp_dof_handler` flag. I mean in terms of a `FECollection` with one fe, we would always refer to it via the zeroth index.

It probably makes sense to introduce an `invalid_fe_index` for internal purposes, but I can't imagine a scenario where we actually need them right now.